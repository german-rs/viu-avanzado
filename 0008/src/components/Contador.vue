<script>
import {mapState} from 'vuex'

export default { 
    computed: {
        ...mapState(['contador']),
        botonObtenerFrutas() {
            this.$store.getters.obtenerFrutas;
        },
    },
    methods:{
        botonIncrementar(){
            this.$store.dispatch('accionIncrementar');
        },
        botonDisminuir(){
            this.$store.dispatch('accionDisminuir');
        }
    }
}
</script>

<template> 
    <div>
        <button @click="botonIncrementar">Incrementar contador</button>
    </div>
    <div>
        <button @click="botonDisminuir">Disminuir contador</button>
    </div>
</template>

<style scoped>
</style>