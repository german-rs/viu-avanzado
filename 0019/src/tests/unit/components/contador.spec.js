import { shallowMount } from '@vue/test-utils' 
import Contador from '@/components/Contador'

//Test 1.
describe('Componente Contador.vue', () => {
  test('Validaci贸n de match con el snapshot', () => {
    //Selecci贸n del sujeto de pruebas
    const wrapper = shallowMount(Contador);

    //Aserci贸n
    expect(wrapper.html()).toMatchSnapshot();
    
    }) 
})

//Test 2.
describe('Componente Contador.vue', () => {
    test('Validar el texto de un bot贸n de enviar formulario', () => { 
        const wrapper = shallowMount(Contador);

        const botonEnviar = wrapper.find('.boton-enviar');

        //Con este console.log comprobamos que estamos accediendo 
        //al texto del elemento
        console.log(botonEnviar.text());

        expect(botonEnviar.text()).toBe("Enviar");
   })
})

//Test 3.
describe('App', () => {
    const wrapper = shallowMount(App);
    it('should have the correct title', () => {
        expect(wrapper.find('h1').text()).toBe('Hello World!');
    }); 
});

//Test 4.
describe('Componente Contador.vue', () => {
    test('Incrementar en 1 el valor del contador', async ()=> {
     const wrapper = shallowMount(Contador)
     const botonIncrementar = wrapper.find('button');
     botonIncrementar.trigger('click')
     const valor = wrapper.find('.incrementar').text()
     expect(valor).toBe('+1')
    })
})