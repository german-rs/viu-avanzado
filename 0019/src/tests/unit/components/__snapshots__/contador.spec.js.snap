// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente Contador.vue Validaci√≥n de match con el snapshot 1`] = `
"<h2 class="titulo-principal">Contador</h2>
<p class="counter">1</p>
<div><button class="btn primary incrementar">+1</button><button class="btn warning disminuir">-1</button><button class="boton-enviar">Enviar</button></div>"
`;
